### PR TITLE
[#36] Implement patternFillCalculator for hatch

### PR DESCRIPTION
**Pull request by kimkanghyune**
- Implementation the ingredient of #36 
- One or more **loops** should be given for constructor
- To draw a single line in hatch region, call `ClipLine` with `line` and this will returns the segments of given `line`
- `Point` is `{ x: number, y: number }`

### CODE REVIEW

File: src/hatch/patternFillCalculator.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+   * 
+   * @param {[Point, Point]} line0 
+   * @param {[Point, Point]} line1 
+   * @returns {number | undefined} t0
+   */
+  _GetIntersection(line0, line1) {
+    const [s0, e0] = line0
+    const [s1, e1] = line1
+    const diff0 = { x: e0.x - s0.x, y: e0.y - s0.y }
+    const diff1 = { x: s1.x - e1.x, y: s1.y - e1.y }
```
Comment (kimkanghyune) [inline src/hatch/patternFillCalculator.js:85]: This is optimized form, solving original equation causes lots of sign invert

**Comment by kimkanghyune** [comment]
Introducing `jest` will be next PR

**Comment by vagran** [comment]
dxf-viewer currently heavily relies on `three.js` for vector math. Shouldn't its `Vector2` class be used here for better readability? `three.js` lacks `Matrix2` but I would introduce one in dxf-viewer (in a separate file), probably will be useful for future features. Just with minimal methods required exactly for this file right now, but compatible in names with `three.js` [`Matrix3`](https://threejs.org/docs/#api/en/math/Matrix3), e.g. , `.invert()`., `.multiply()`, etc.

**Comment by Phryxia** [comment]
> dxf-viewer currently heavily relies on `three.js` for vector math. Shouldn't its `Vector2` class be used here for better readability? `three.js` lacks `Matrix2` but I would introduce one in dxf-viewer (in a separate file), probably will be useful for future features. Just with minimal methods required exactly for this file right now, but compatible in names with `three.js` [`Matrix3`](https://threejs.org/docs/#api/en/math/Matrix3), e.g. , `.invert()`., `.multiply()`, etc.
> 
> If you feel it currently does not worth the time, we can leave it as is. I can polish it later.

@vagran
I'm the author of this PR (=@kimkanghyune) 
I'll do that but this week I can't proceed due to my company's situation, so please wait for modification. Thank you for your kind feedback and patience!

**Comment by kimkanghyune** [comment]
@vagran 

> Shouldn't its Vector2 class be used here for better readability?

Polishing is finished! Please verify the changes.

**Pull request merged**